capstone/src/main/java/com/google/step/servlets/RestaurantServlet.java

```diff
@@ -0,0 +1,74 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.step.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.GeoPt;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.gson.Gson;
+import com.google.gson.JsonObject;
+import com.google.step.data.Restaurant;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet responsible for sending/getting restaurants to/from Datastore. */
+@WebServlet("/restaurant")
+public class RestaurantServlet extends HttpServlet {
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
```

Comment:
Please add a method comment. Readers might assume that a get on a "restaurant" servlet only returns a single restaurant, so it's nice to specify in the method comment what is returned.

Reply:
Awesome, I added method comments in the new servlets for listing restaurants versus requesting just one.

Reply:
Thanks! Would you mind also describing the request format, the response format, and any error codes returned (for example not found). Basically after reading the method comment, readers should know exactly how to call the method and how to process the result.

For specifying JSON objects you can do something like this (example reused from another code review):
@param response The login status and the url for changing the login status, in the following json format:
{
  "loggedIn": \<boolean\>,
  "email": \<string, only specified if loggedIn is true\>,
  "logoutURL": \<string, only specified if loggedIn is true\>,
  "loginURL": \<string, only specified if loggedIn is false\>,
}